jbehave-support-core/src/main/java/org/jbehavesupport/core/XmlReporterAutoConfiguration.java

```diff
@@ -0,0 +1,164 @@
+import org.springframework.boot.autoconfigure.condition.NoneNestedConditions;
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ConfigurationCondition;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.EnumerablePropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+import org.springframework.util.StringUtils;
+
+import javax.sql.DataSource;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Arrays;
+import java.util.Map;
+
+@Configuration
+public class XmlReporterAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnProperty(name = "reporting.enabled", havingValue = "on", matchIfMissing = true)
+    public XmlReporterFactory xmlReporterFactory() {
+        return new XmlReporterFactory();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnBean(XmlReporterFactory.class)
+    @ConditionalOnProperty(name = "reporting.context.enabled", havingValue = "on", matchIfMissing = true)
+    public TestContextXmlReporterExtension testContextXmlReporterExtension(TestContext testContext) {
+        return new TestContextXmlReporterExtension(testContext);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnBean(XmlReporterFactory.class)
+    @ConditionalOnProperty(name = "reporting.web.screenshot.failed.enabled", havingValue = "on", matchIfMissing = true)
+    public FailScreenshotsReporterExtension failScreenshotsReporterExtension(TestContext testContext) {
+        return new FailScreenshotsReporterExtension(testContext);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnBean(XmlReporterFactory.class)
+    @Conditional(ScreenshotReporterCondition.class)
+    public ScreenshotReporterExtension screenshotReporterExtension(TestContext testContext) {
+        return new ScreenshotReporterExtension(testContext);
+    }
+
+    @Bean
+    @ConditionalOnBean(XmlReporterFactory.class)
+    @ConditionalOnMissingBean
+    @ConditionalOnPropertyPrefix(prefix = "environmentInfo")
+    public EnvironmentInfoXmlReporterExtension environmentInfoXmlReporterExtension(Environment environment) {
+        return new EnvironmentInfoXmlReporterExtension(environment);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnBean({RestServiceHandler.class, XmlReporterFactory.class})
+    public RestXmlReporterExtension restXmlReporterExtension(TestContext testContext, FileNameResolver fileNameResolver) {
```

Comment:
I would add conditional on property with default true, to give users option to turn it off.

Reply:
Done, I have added several (hopefully havent forgotten any)